pkg/storage/segments/store.go

```diff
@@ -339,9 +344,9 @@ func (s *segmentStore) Repair(ctx context.Context, path storj.Path, lostPieces [
 	totalRepairCount := len(newNodes)
 
 	//make a repair nodes list just with new unique ids
-	repairNodesList := make([]*pb.Node, len(originalNodes))
-	for j, vr := range originalNodes {
-		// find the nil in the original node list
+	repairNodesList := make([]*pb.Node, len(healthyNodes))
+	for j, vr := range healthyNodes {
+		// find the nil in the node list
 		if vr == nil {
 			// replace the location with the newNode Node info
 			totalRepairCount--
```

Comment:
Another sanity check that `totalRepairCount` is non-negative.

Then after the for-loop finishes, we expect that `totalRepairCount == 0`. Let's check for it too.